### PR TITLE
Upgrade to PowSyBl dependencies 2022.0.0

### PR DESCRIPTION
**Pull request by OpenSuze**
Signed-off-by: Hugo SCHINDLER <hugo.schindler@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
Add sensitivity batch size parameter
Move Xnec Selection outside of flow decomposition algorithm

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: java/pom.xml [hunk 1]
```diff
@@ -74,7 +74,7 @@
         <maven-dependency-plugin.version>3.2.0</maven-dependency-plugin.version>
         <maven-compiler-plugin.version>3.10.1</maven-compiler-plugin.version>
         <maven-shade-plugin.version>3.2.4</maven-shade-plugin.version>
-        <powsybl-dependencies.version>1.3.1</powsybl-dependencies.version>
+        <powsybl-dependencies.version>1.4.0-SNAPSHOT</powsybl-dependencies.version>
```
Comment (EtienneLt) [inline java/pom.xml]: could you use integration branch please ?

Comment (EtienneLt) [reply java/pom.xml]: maybe it needs to be updated

Comment (OpenSuze) [reply java/pom.xml]: rebased to integration branch :)

Comment (OpenSuze) [reply java/pom.xml]: I removed this integration for CI tests

File: java/src/main/java/com/powsybl/python/flow_decomposition/FlowDecompositionContext.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.powsybl.python.flow_decomposition;
```
Comment (murgeyseb) [inline java/src/main/java/com/powsybl/python/flow_decomposition/FlowDecompositionContext.java:8]: Copyright header

File: java/src/main/java/com/powsybl/python/flow_decomposition/FlowDecompositionContext.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
```
Comment (murgeyseb) [inline java/src/main/java/com/powsybl/python/flow_decomposition/FlowDecompositionContext.java:12]: Author header

File: tests/test_flow_decomposition.py [hunk 4]
```diff
@@ -93,6 +115,8 @@ def test_flow_decomposition_parameters():
     logging.getLogger('powsybl').setLevel(logging.DEBUG)
     test_demo()
     print('\n\n\n\n')
+    test_demo_one_by_one()
+    print('\n\n\n\n')
```
Comment (EtienneLt) [inline tests/test_flow_decomposition.py]: a little thing could you remove all the print in this file ?

**Comment by geofjamg** [comment]
I upgraded to powsybl dependencies 2.0.0 and merge some of the modification of #499.
Some unit test are not passing because adaptation for security analysis status PR has not been done et still is to be done. The old isComputationOk is not working anymore and for pre contingency calculation always return ok... 
We need to adapt the code to get the status which is at the PreContingencyResult and PostContingencyResult class.

**Comment by geofjamg** [comment]
@EtienneLt @sylvlecl is one of you available to work on contingency status?

**Comment by EtienneLt** [comment]
> @EtienneLt @sylvlecl is one of you available to work on contingency status?

yes I can do that

File: java/src/main/java/com/powsybl/python/flow_decomposition/FlowDecompositionContext.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
```
Comment (EtienneLt) [inline java/src/main/java/com/powsybl/python/flow_decomposition/FlowDecompositionContext.java:7]: can you add the SPDX code
```java
/**
 * Copyright (c) 2022, RTE (http://www.rte-france.com)
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 *  SPDX-License-Identifier: MPL-2.0
 */
```

File: java/src/main/java/com/powsybl/python/security/SecurityAnalysisCFunctions.java [hunk 6]
```diff
@@ -164,14 +181,20 @@ private static void setSecurityAnalysisResultPointer(PyPowsyblApiHeader.Continge
         contingencyPtr.limitViolations().setPtr(limitViolationPtr);
     }
 
-    private static PyPowsyblApiHeader.ArrayPointer<PyPowsyblApiHeader.ContingencyResultPointer> createContingencyResultArrayPointer(SecurityAnalysisResult result) {
+    private static PyPowsyblApiHeader.PreContingencyResultPointer createPreContingencyResultArrayPointer(SecurityAnalysisResult result) {
+        int resultCount = result.getPostContingencyResults().size() + 1; // + 1 for pre-contingency result
```
Comment (geofjamg) [inline java/src/main/java/com/powsybl/python/security/SecurityAnalysisCFunctions.java]: just one?

File: java/src/main/java/com/powsybl/python/security/SecurityAnalysisCFunctions.java [hunk 7]
```diff
@@ -138,14 +138,31 @@ public static void addContingency(IsolateThread thread, ObjectHandle contingency
         });
     }
 
-    private static LoadFlowResult.ComponentResult.Status getStatus(LimitViolationsResult result) {
-        return result.isComputationOk() ? LoadFlowResult.ComponentResult.Status.CONVERGED : LoadFlowResult.ComponentResult.Status.FAILED;
+    private static void setPostContingencyResultInSecurityAnalysisResultPointer(PyPowsyblApiHeader.PostContingencyResultPointer contingencyPtr, PostContingencyResult postContingencyResult) {
+        contingencyPtr.setContingencyId(CTypeUtil.toCharPtr(postContingencyResult.getContingency().getId()));
+        contingencyPtr.setStatus(postContingencyResult.getStatus().ordinal());
+        List<LimitViolation> limitViolations = postContingencyResult.getLimitViolationsResult().getLimitViolations();
```
Comment (geofjamg) [inline java/src/main/java/com/powsybl/python/security/SecurityAnalysisCFunctions.java:144]: Maybe we could create a function to share limit violations code between pre and post contingency?

File: java/src/main/java/com/powsybl/python/security/SecurityAnalysisCFunctions.java (hunk 6 above)
Comment (EtienneLt) [reply java/src/main/java/com/powsybl/python/security/SecurityAnalysisCFunctions.java]: I forget to remove it now that they are separated

**Review by geofjamg — APPROVED** [review]

File: tests/test_flow_decomposition.py (hunk 4 above)
Comment (OpenSuze) [reply tests/test_flow_decomposition.py]: done

**Review by OpenSuze — COMMENTED** [review]
Ready for merging. Everything looks ok on my side :) Thanks for the update !

**Review by EtienneLt — APPROVED** [review]

**Pull request merged**